wgpu-rs/src/lib.rs

```diff
@@ -455,10 +456,46 @@ impl Device {
     }
 }
 
+struct BufferMapReadAsyncUserdata {
```

Comment:
we should make this generic over the function, to avoid double boxing (one here, one in `let userdata = ...`)